packages/main/src/plugin/kubernetes/contexts-manager.ts

```diff
@@ -965,7 +882,11 @@ export class ContextsManager {
       return [];
     }
     console.debug(`start watching ${resourceName} in context ${this.currentContext}`);
-    this.startResourceInformer(this.currentContext.name, resourceName);
+    // start secondary informers only if current context is reachable
+    const currentContextState = this.states.getCurrentContextGeneralState(this.currentContext.name);
+    if (currentContextState.reachable) {
```

Comment:
I'm having trouble following the async states - how do we know the reachable state of the current context will be valid / updated when this function is called, and isn't that basically the same as the call to .isReachable() a few lines up?

Reply:
yes, you are right this is the same, I'll change to simplify